mrp_subcontracting_bom_dual_use/models/mrp_bom.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class MrpBom(models.Model):
+    _inherit = "mrp.bom"
+
+    allow_in_regular_production = fields.Boolean(string="Allow in regular production")
+
+    @api.model
+    def _bom_find_domain(
+        self,
+        product_tmpl=None,
+        product=None,
+        picking_type=None,
+        company_id=False,
+        bom_type=False,
+    ):
+        """We need to overwrite the domain to share boms"""
+        domain = super(MrpBom, self)._bom_find_domain(
+            product_tmpl=product_tmpl,
+            product=product,
+            picking_type=picking_type,
+            company_id=company_id,
+            bom_type=False,
```

Comment:
is this on purpose?

Reply:
Yes, the previous explanation is not too extensive but it tries to clarify it.

By default, odoo auto-adds the domain to filter by type but we want to remote it to apply after the type filter in a custom way and be able to use type subcontracting in manufacturing.

Reply:
This is not correct, as kit BoMs will be considered as well

Reply:
Ok, I'll clarify it more.

1. We will apply the same filter as Odoo if `bom_type` had been defined.

`if bom_type:
             new_domain = [("type", "=", bom_type)]`

2. If bom_type = normal (manufacturing) we will add the following:

`new_domain = expression.OR(
                     [
                         new_domain,
                         [
                             ("type", "!=", bom_type),
                             ("allow_in_regular_production", "=", True),
                         ],
                     ]
                 )
`

3. We will add the new domain:

`domain = expression.AND([domain, new_domain])`